### DISCUSSION TITLE
Confusion over Step 4 and bin/winapps check while using docker container

### DISCUSSION THREAD

**Discussion by WarMuffins** [winapps-org/winapps#137] [Support] [answered]
Hello.

I've been following the installation instructions pretty smoothly so far and have a docker vm (or is the proper term "container"?) set up. However, when I reached step 4 and was required to run bin/winapps check I got this error

``RDPWindows is not running, run:
  virsh start RDPWindows``
  
  running ./installer.sh also results in a similar error
  
  `RDPWindows is not running, run:
  virsh start RDPWindows`
  
  I am somewhat familiar with virsh, mainly through virtual machine manager, but completely new to docker. I'm pretty sure virsh isn't involved with docker at all, but I could be wrong. Regardless, I do not know how to get the winapps and the installer to "see" my windows install with docker.
  
  Did I miss something in the docker instructions? Right now I am redoing the install following the KVM.md instructions- however I prefer to use docker if that's apparently where things are headed for this project.
  
  Thanks.

**Comment by oskardotglobal** [comment]
You're missing the RDP_IP in the config

**Reply by WarMuffins** [reply]
I noticed this in the instructions after you pointed that out to me.
> When using a pre-existing non-KVM RDP server, you must use RDP_IP to specify the location of the Windows server.

Which IP should I use? So far there's the IP that I use to access the container on my browser 127.0.0.1:8006 (however I don't think I can include the port, so I would just be providing a loopback address) or the address given by running ``docker inspect <container id> | grep "IPAddress"`` which gives me something else. Putting both of them into``~/.config/winapps/winapps.conf`` give the same error when running ``bin/winapps check`` 

> bin/winapps: line 97: /d:: No such file or directory

**Reply by LDprg** [reply, marked as answer]
@WarMuffins you should use 127.0.0.1

The docker ip is used for local networks of containers. They won't work for this application.

**Reply by WarMuffins** [reply]
That seemed to have worked, however now my issue has something to do with freerdp3-x11. Since freerdp3-x11 isn't available on my package manager, I have installed the Flatpak. However, it seems like the command for the flatpak is different and the winapps script isn't running it.

When I try to manually type "xfreerdp" I get a command not found and a prompt to apt install freerdp2-x11.

Installing that and running winapps check does result in some further execution (and it also seemed that it was able to recognize the docker container) however it eventually fails, which is expected since rdp2 is out of date for winapps. (I immediately apt purged it to clean up). 


I noticed the ``FREERDP_COMMAND="xfreerdp"`` configuration in the config file, what should I put in there to be able to run the flatpak? Or did I install the flatpak incorrectly? I am mostly new to flatpaks.

**Reply by oskardotglobal** [reply]
As I have detailed on #117 before, you can use a wrapper script for flatpak freerdp and point winapps to that file.
```
#!/usr/bin/env bash
flatpak run --command=xfreerdp com.freerdp.FreeRDP "$@"
```
Put this file in any directory and make sure to make it executable. Then, set `FREERDP_COMMAND` to the path to that file.

**Reply by KernelGhost** [reply]
@WarMuffins The repository has been updated to support Flatpak installations of `FreeRDP`. Please read the updated project README. There should no longer be a need to create your own wrapper script.

**Reply by WarMuffins** [reply]
Everything seems to be going good now! Aside from the installer not being able to recognize the docker container.

Friendly reminder to anyone reading this in the future, make sure to rename the docker install to WinApps

**Comment by oskardotglobal** [comment]
The error message is misleading

**Reply by KernelGhost** [reply]
Given the recent changes, this error message should no longer appear unless the user has specified they are using `libvirt` within their `~/.config/winapps/winapps.conf`.
